src/unleash-proxy.ts

```diff
@@ -261,109 +271,83 @@ export default class UnleashProxy {
         this.clientKeys = clientKeys;
     }
 
-    getAllToggles(req: Request, res: Response<FeaturesSchema | string>): void {
+    private readyMiddleware(req: Request, res: Response, next: NextFunction) {
         const apiToken = req.header(this.clientKeysHeaderName);
 
-        if (!this.enableAllEndpoint) {
-            res.status(501).send(
-                'The /proxy/all endpoint is disabled. Please check your server configuration. To enable it, set the `enableAllEndpoint` configuration option or `ENABLE_ALL_ENDPOINT` environment variable to `true`.',
-            );
-        } else if (!this.ready) {
+        if (!this.ready) {
             res.status(503).send(NOT_READY_MSG);
         } else if (!apiToken || !this.clientKeys.includes(apiToken)) {
             res.sendStatus(401);
         } else {
-            const { query } = req;
-            query.remoteAddress = query.remoteAddress || req.ip;
-            const context = createContext(query);
-            const toggles = this.client.getAllToggles(context);
-            res.set('Cache-control', 'public, max-age=2');
-            res.send({ toggles });
+            next();
         }
     }
 
-    getAllTogglesPOST(
+    async getAllToggles(
```

Comment:
I guess nothing changes when switching to `async`, express supports this, but since we're using express v4 there's this caveat about error handling: https://expressjs.com/en/guide/error-handling.html 

> For errors returned from asynchronous functions invoked by route handlers and middleware, you must pass them to the next() function, where Express will catch and process them


I checked the code, and all expected errors are handled by using `res.sendStatus` rather than throwing an exception, meaning we shouldn't worry about this, but I wonder if we should have some global try-catch that would call `next(err)` if something unexpected happens.

Reply:
But are these methods even async now? Everything async is handled within the `createContexMiddleware`, if I'm not mistaken, so the methods down here could be kept without `async`. However, catching errors from `await enrichContext` might be reasonable, as the context enrichers are user-defined and could fail.